src/iris_api/bin/sender.py

```diff
@@ -808,6 +808,14 @@ def distributed_send_message(message):
 
 def fetch_and_send_message():
     message = send_queue.get()
+
+    if not cache.application_quotas(message):
```

Comment:
cache seems like a weird namespace for this. It's real SoT data in memory, not cached data.